### PR TITLE
winch(arm64): and, or, xor, shifts

### PR DESCRIPTION
**Pull request by evacchi**
Follow up to #8321.

Adds support to binary and, binary or, binary xor (eor), shl, shrS, shrU, rotr, rotl.

### CODE REVIEW

**Comment by saulecabrera** [comment]
I'll take a look at this one.

File: winch/codegen/src/masm.rs
Comment (saulecabrera) [inline winch/codegen/src/masm.rs:242]: A conversion [already exists](https://github.com/bytecodealliance/wasmtime/blob/main/winch/codegen/src/isa/aarch64/asm.rs#L22) between Winch's `OperandSize` and Cranelift's aarch64 `OperandSize`; I believe we could get drop this method and instead do something like

```rust
size.into().to_ty()
```

File: winch/codegen/src/isa/aarch64/asm.rs
Comment (evacchi) [reply winch/codegen/src/isa/aarch64/asm.rs:43]: refactored and moved to `shift_kind_to_alu_op()`

File: winch/codegen/src/masm.rs
Comment (evacchi) [reply winch/codegen/src/masm.rs]: done!

File: winch/codegen/src/isa/aarch64/masm.rs
Comment (evacchi) [reply winch/codegen/src/isa/aarch64/masm.rs:439]: I think I got it, will do

Comment (evacchi) [reply winch/codegen/src/isa/aarch64/masm.rs]: I hope I got it right 😬

File: winch/codegen/src/codegen/mod.rs
Comment (evacchi) [inline winch/codegen/src/codegen/mod.rs:861]: I hope these are correct 😬

**Review by saulecabrera — APPROVED** [review]
This looks good to me, thanks!

For visibility, here's a recap of an offline discussion between @evacchi and myself:

* @evacchi is on PTO for the next couple of weeks and therefore suggested pushing any required changes on top of this PR
* I pushed https://github.com/bytecodealliance/wasmtime/pull/8921/commits/e18681c3b716aa85b9361f20b9926fcb28b06bd1, which is a mechanical change only. 

Given the recent changes I was thinking it might make sense to get a second review on this change.

@elliottt, would you be able to provide a second pass on this change?

**Review by elliottt — APPROVED** [review]
This looks good to me as well!

**Comment by evacchi** [comment]
Wonderful thanks for pushing this to the finish line!! ❤️❤️

**Pull request merged**
